File: app/src/main/java/in/sudopk/vaishnavacalendar/retrofit/VcService.java
package in.sudopk.vaishnavacalendar.retrofit;

import java.util.List;

import in.sudopk.vaishnavacalendar.Country;
import in.sudopk.vaishnavacalendar.VCalendar;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface VcService {
    String URL =  "http://www.vaisnavacalendar.com/";

    @GET("vcal.php")
    @VcApi.Calendar
    Call<VCalendar> calendar(@Query("month") String month, @Query("year") int year,
                                       @Query("lang") String lang, @Query("CIID") String locationId);

    @GET("vcal.php")
    @VcApi.Locations
    Call<List<Country>> locations();
}

File: app/src/main/java/in/sudopk/vaishnavacalendar/retrofit/VcService.kt
package `in`.sudopk.vaishnavacalendar.retrofit

import `in`.sudopk.vaishnavacalendar.calendar.Country
import `in`.sudopk.vaishnavacalendar.VCalendar
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.http.GET
import retrofit2.http.Query

interface VcService {

    @GET("vcal.php")
    @VcApi.Calendar
    fun calendar(@Query("month") month: String, @Query("year") year: Int,
                 @Query("lang") lang: String, @Query("CIID") locationId: String): Call<VCalendar>

    @GET("vcal.php")
    @VcApi.Locations
    fun locations(): Call<List<Country>>

    companion object {
        val URL = "http://www.vaisnavacalendar.com/"

        fun newInstance(): VcService {
            return Retrofit.Builder()
                    .baseUrl(VcService.URL)
                    .addConverterFactory(VcConverterFactory())
                    .build()
                    .create(VcService::class.java)
        }
    }
}
